### ISSUE TITLE
Artwork caching issues

### ISSUE THREAD

**Issue by croneter** [croneter/PlexKodiConnect#461]
Artwork caching is not working as expected:
* Seems to have started with PKC beta version 2.0.5, 2 months ago
* Issues seem related to using LOCAL PMS; issue not reproducable using remote PMS
* TBV: Deactivating artwork caching seems to solve the issue

**Comment by MacQwerty** [comment]
I've the same problem with CoreELEC (official): 8.90.2 (9.0), kernel: Linux ARM 64-bit version 3.14.29 aarch64, Kodi (18.0-ALPHA2 Git:3a989ee) and PKC 2.0.22. 

-`ERROR: CCurlFile::Stat - Failed: HTTP response code said error(22) for https://assets.fanart.tv/fanart/movies/X/moviethumb/X.jpg`
-`DEBUG: CCurlFile::GetMimeType - https://assets.fanart.tv/fanart/movies/X/moviethumb/X.jpg -> failed`
-`DEBUG: CDirectoryProvider[special://skin/playlists/recent_unwatched_movies.xsp]: refreshing..`
-`DEBUG: CDirectoryProvider[videodb://movies/genres/]: refreshing..`
-`DEBUG: GLES: Extension Support Test - GL_EXT_texture_format_BGRA8888 YES`
-`DEBUG: CCurlFile::GetMimeType - https://assets.fanart.tv/fanart/movies/X/moviethumb/X.jpg -> image/jpeg`
-`DEBUG: PLEX.librarysync: self.updatelist: []`
-`NOTICE: PLEX.librarysync: Movies sync is finished.`
-`NOTICE: PLEX.utils: It took 0:00:00.805378 to run the function plex_movies`
-`NOTICE: PLEX.librarysync: Media folders for TVShows: [{'itemtype': u'show', 'kodi_tagid': 2, 'id': u'2', 'sync_to_kodi': 1, 'name': u'Series'}]`
-`DEBUG: PLEX.librarysync: Analyzed view Series with ID 2`
-`DEBUG: PLEX.librarysync: self.updatelist: []`
-`DEBUG: PLEX.librarysync: process_updatelist completed for tv shows`
-`DEBUG: PLEX.librarysync: Analyzed all seasons of TV show with Plex Id X`
-`DEBUG: CCurlFile::GetMimeType - https://assets.fanart.tv/fanart/movies/X/moviethumb/X.jpg -> image/jpeg`
-`DEBUG: PLEX.librarysync: Analyzed all seasons of TV show with Plex Id X`
-`DEBUG: CCurlFile::GetMimeType - https://IP:32400/photo/:/transcode?width=4000&height=4000&minSize=1&upscale=0&url=/library/metadata/X/thumb/X-Plex-Token=X -> image/jpeg`
-`DEBUG: CCurlFile::GetMimeType - https://IP:32400/photo/:/transcode?width=4000&height=4000&minSize=1&upscale=0&url=/library/metadata/X/thumb/X-Plex-Token=X -> image/jpeg`

**Comment by hanzoh** [comment]
I have now disabled HTTPS in PKC and everything is working again (with caching and additional artwork enabled).

**Comment by metzger2309** [comment]
@hanzoh, I fear Kodi will download its own artwork from the internet if you enable additional artwork from fanart. I think it will not catch it from your plex server, but from the internet.

**Comment by croneter** [comment]
I cannot reproduce this issue anymore, even though I could do so last week. My guess: it was caused on the Plex Media Server side of things. You all please make sure that you're using the latest PMS version >=1.13.0.5023. A PMS restart might also help :wink: 

I **do** have secure HTTPS connections enabled, even required (as everyone should have). Everything is working fine even though I'm using local LAN network IPs:
```
DEBUG: Caching image 'https://192.168.0.2:32400/photo/:/transcode?width=4000&height=4000&minSize=1&upscale=0&url=/library/metadata/541310/art/1525440070&X-Plex-Token=XXXX' to '6/69a147f6.jpg':
DEBUG: cached image 'special://masterprofile/Thumbnails/6/69a147f6.jpg' size 1920x1080
```

![unbenannt](https://user-images.githubusercontent.com/4637940/39965967-de60e484-56a3-11e8-9df6-98cbfc45a602.PNG)

I do **not** get any errors like @hanzoh 
```
ERROR: CCurlFile::Stat - Failed: SSL peer certificate or SSH remote key was not OK(51) for 
```

Thanks for reporting back you guys whether this is still an issue!

**Comment by croneter** [comment]
@metzger2309 , @hanzoh 
If you have additional artwork turned on in the PKC settings under `Artwork`, yes, additional artwork will be downloaded from the internet. However, everything should be working fine (Plex/PMS artwork as well as additional artwork). Thanks for reporting back!

**Comment by croneter** [comment]
@MacQwerty
Did you update your PMS t I the latest version? Did you restart your PMS at least once?

**Comment by danjames92** [comment]
I've only started to see this since the latest beta, 2.0.23

Remote server (plex through Cloudflare)
HTTPS via port 443.

2.0.22 seemed to work fine.

**Comment by dazedcrazy** [comment]
Plex Version 1.13.0.5023
PKC Version 2.0.23
Kodi is latest nightly
Debug:  [kodi.log](https://github.com/croneter/PlexKodiConnect/files/2001463/kodi.log)
This is been happening for a while in Leia, I thought it was related to [this](https://forum.kodi.tv/showthread.php?tid=331124) but that was fixed. Then thought it was smb/network access thing but this works fine
`16:21:50.512 T:11256   DEBUG: Caching image 'smb://USERNAME:PASSWORD@192.168.0.4/movie/[movieset-art]/Back to the Future Collection/poster.jpg' to '5/53a3e1e8.jpg':
16:21:50.512 T:11256   DEBUG: cached image 'special://masterprofile/Thumbnails/5/53a3e1e8.jpg' size 1000x1500`

**Comment by croneter** [comment]
Just to check here, @dazedcrazy and @MacQwerty , playback DOES work for addon paths? If you configured direct paths, you can use the Plex context menu and then choose `Start playback via PMS` to force addon path playback

I'll take drastic measures here to completely avoid SSL issues with Kodi: PKC will use plex.direct URLs instead of the direct IPs (like 192.168.1.2) it has been relying on until now

**Comment by MacQwerty** [comment]
@croneter , Yeah, it works perfectly! I actually did a 300 mbit Jellyfish test yesterday without issues! So the only problem I've encountered during the months I've been using PKC with addon paths is the SSL - artwork issue... Which is a minor bagatelle since SSL can be turned off...

**Comment by croneter** [comment]
Cool! So is artwork caching now fixed with 2.0.24?
